android/BOINC/app/src/main/java/edu/berkeley/boinc/rpcExtern/RpcSettingsData.kt

```diff
@@ -0,0 +1,188 @@
+package edu.berkeley.boinc.rpcExtern
+
+import android.os.Parcel
+import android.os.Parcelable
+import kotlinx.android.parcel.Parcelize
+
+class RpcSettingsDatax {
+    val DEFAULT_PORT = 31416
+    val data = RpcSettingsData()
+    var validData = false
+    var ipAllowedList = ArrayList<String>()
+
+    fun set(enabled : Boolean, encryption : Boolean, passwrd : String, port :String, ip1 : String, ip2 : String, ip3 : String , ip4 : String)
+    {
+        validData = true
+        data.externEnabled = enabled
+        data.externEncryption = encryption
+        data.externPasswrd = passwrd
+
+        try {
+            if (port.isNotEmpty()) {
+                data.externPort = port.toInt()
+            }
+            else
+            {
+                data.externPort = DEFAULT_PORT
+            }
+        } catch (e: Exception) {
+            data.externPort = DEFAULT_PORT
+        }
+
+        data.ipAllowed1 = ip1
+        data.ipAllowed2 = ip2
+        data.ipAllowed3 = ip3
+        data.ipAllowed4 = ip4
+
+        if (data.ipAllowed1.isNotEmpty()) {
+            ipAllowedList.add(data.ipAllowed1)
+        }
+        if (data.ipAllowed2.isNotEmpty()) {
+            ipAllowedList.add(data.ipAllowed2)
+        }
+        if (data.ipAllowed3.isNotEmpty()) {
+            ipAllowedList.add(data.ipAllowed3)
+        }
+        if (data.ipAllowed4.isNotEmpty()) {
+            ipAllowedList.add(data.ipAllowed4)
+        }
+    }
+}
+
+class  RpcSettingsData() : Parcelable {
```

Comment:
The variables can be declared as part of the constructor.

Reply:
I know but this worked more stable and the effect if probably similar, but hidden.

Reply:
To clarify I used the single Parcelize method, but this gave erratic results, this works fine.